main.js

```diff
@@ -0,0 +1,33 @@
+const { tBodies } = document.getElementById('table');
+const xhrUsers = new XMLHttpRequest();
+const xhrPosts = new XMLHttpRequest();
+let users;
+let posts;
+
+document.addEventListener('DOMContentLoaded', () => {
+    xhrUsers.open("GET", 'https://jsonplaceholder.typicode.com/users');
+    xhrPosts.open("GET", ' https://jsonplaceholder.typicode.com/todos');
+    xhrUsers.send();
+    xhrUsers.responseType = 'json';
+
+    xhrUsers.addEventListener('load', function() {
+        users = xhrUsers.response;
+        xhrPosts.send();
+        xhrPosts.responseType = 'json';
+
+        xhrPosts.addEventListener('load', function() {
+            posts = xhrPosts.response;
+
+            posts.forEach(post => {
+                const { userId } = post;
+                const user = users.find(item => item.id === userId);
+
+                tBodies[0].insertAdjacentHTML('beforeend', `

```

Comment:
the solution looks great(except formatting html) and compact, but... do not use innerHTML(insertAdjacentHTML - is mostly the same), because you insert unknown data, it is not secure
https://uk.wikipedia.org/wiki/%D0%9C%D1%96%D0%B6%D1%81%D0%B0%D0%B9%D1%82%D0%BE%D0%B2%D0%B8%D0%B9_%D1%81%D0%BA%D1%80%D0%B8%D0%BF%D1%82%D0%B8%D0%BD%D0%B3

https://developer.mozilla.org/en-US/docs/Glossary/Cross-site_scripting